File: DIYHomeAutomation/app/src/main/java/com/example/diyhomeautomation/models/Notification.kt
package com.example.diyhomeautomation.models

import java.util.Date

/**
 * Data class representing notifications related to tasks and devices.
 *
 * @property id The unique identifier for the notification.
 * @property name The name of the notification.
 * @property description The description of the notification.
 * @property time The time the notification occurred.
 * @property visualizeState Indicates whether the notification should be visually represented.
 * @property taskId The ID of the associated task.
 * @property deviceId The ID of the associated device.
 *
 * @constructor Creates an instance of Notification with the provided properties.
 */
data class Notification(
    var id: Int,
    var name: String,
    var description: String,
    var time: Date,
    var visualizeState: Boolean,
    var taskId: Int,
    var deviceId: Int
)

File: DIYHomeAutomationAPI/DIYHomeAutomationAPI/Models/Notification.cs
using DIYHomeAutomationAPI.Models.Enums;

namespace DIYHomeAutomationAPI.Models
{
    /// <summary>
    /// This class represents a Notification
    /// </summary>
    public class Notification
    {
        /// <summary>
        /// Notification's Id
        /// </summary>
        public int Id { get; set; }

        /// <summary>
        /// Notification's Name
        /// </summary>
        public string Name { get; set; } = null!;

        /// <summary>
        /// Notification's Description
        /// </summary>
        public string Description { get; set; } = null!;

        /// <summary>
        /// Notification's Date
        /// </summary>
        public DateTime Time { get; set; }

        /// <summary>
        /// Notification's visualization state
        /// </summary>
        public bool VisualizeState { get; set; }

        /// <summary>
        /// Task's Id
        /// </summary>
        public int? TaskId { get; set; }

        /// <summary>
        /// Device's Id
        /// </summary>
        public int? DeviceId { get; set; }

    }
}
